server/app/mutations/grid_services/delete.rb

```diff
@@ -7,18 +7,39 @@ class Delete < Mutations::Command
     end
 
     def validate
-      if self.grid_service.deploying?
-        add_error(:service, :invalid, "Cannot delete service because it's currently being deployed")
-        return
-      end
       linked_from_services = self.grid_service.linked_from_services
       if linked_from_services.count > 0
         add_error(:service, :invalid, "Cannot delete service that is linked to another service (#{linked_from_services.map{|s| s.name}.join(', ')})")
       end
     end
 
     def execute
-      worker(:grid_service_remove).async.perform(self.grid_service.id)
+      nodes = self.grid_service.grid_service_instances.map{ |i| i.host_node }
+      notify_lb_remove if self.grid_service.linked_to_load_balancer?
```

Comment:
Hmm... if the service is in a restart loop, and restarts after this removal, but before the agent kills it, then the service will be re-added to etcd?